### PR TITLE
Opr 163 bugfix6

### PR DESCRIPTION
**Pull request by LiekeSchoenmaker**

### CODE REVIEW

File: patient-reporter/src/main/java/com/hartwig/oncoact/patientreporter/cfreport/chapters/analysed/ClinicalEvidenceFunctions.java [hunk 1]
```diff
@@ -83,9 +87,10 @@ public static Map<String, List<ProtectEvidence>> buildTreatmentMap(@NotNull List
 
         evidences.sort(Comparator.comparing(ProtectEvidence::event)
                 .thenComparing(it -> it.gene() != null ? it.gene() : Strings.EMPTY)
+                .thenComparing(ProtectEvidence::onLabel)
+                .reversed()
```
Comment (mvanniekerkHartwig) [inline patient-reporter/src/main/java/com/hartwig/oncoact/patientreporter/cfreport/chapters/analysed/ClinicalEvidenceFunctions.java:91]: Are you sure this does not convert the whole comparator but only this last`thenComparing` step?

**Review by mvanniekerkHartwig — COMMENTED** [review]
It would be nice if we could simplify this code. I heard from Mathijs you worked on a new implementation that is supposed to be significantly simpler. What is the status of that?

File: patient-reporter/src/main/java/com/hartwig/oncoact/patientreporter/cfreport/chapters/analysed/ClinicalEvidenceFunctions.java (hunk 1 above)
Comment (LiekeSchoenmaker) [reply patient-reporter/src/main/java/com/hartwig/oncoact/patientreporter/cfreport/chapters/analysed/ClinicalEvidenceFunctions.java:91]: I have logged the output and this looks like only the onLabel match

**Comment by LiekeSchoenmaker** [comment]
About simplify the code; I will first discuss the total logica before we go to simplify the code. This will be improved in a next version

File: patient-reporter/src/test/java/com/hartwig/oncoact/patientreporter/cfreport/chapters/analysed/ClinicalEvidenceFunctionsTest.java [hunk 2]
```diff
@@ -39,26 +38,7 @@ public void canDetermineHigherEvidence() {
...
-                EvidenceDirection.RESISTANT,
-                true), createProtectEvidence(EvidenceLevel.A, EvidenceDirection.RESISTANT, true)));
-        assertTrue(ClinicalEvidenceFunctions.hasHigherOrEqualEvidenceForEventAndTreatment(createProtectEvidenceList(EvidenceLevel.A,
-                EvidenceDirection.RESPONSIVE,
-                true), createProtectEvidence(EvidenceLevel.B, EvidenceDirection.RESPONSIVE, true)));
-        assertTrue(ClinicalEvidenceFunctions.hasHigherOrEqualEvidenceForEventAndTreatment(createProtectEvidenceList(EvidenceLevel.A,
-                EvidenceDirection.PREDICTED_RESPONSIVE,
-                true), createProtectEvidence(EvidenceLevel.C, EvidenceDirection.RESPONSIVE, true)));
-        assertFalse(ClinicalEvidenceFunctions.hasHigherOrEqualEvidenceForEventAndTreatment(createProtectEvidenceList(EvidenceLevel.C,
+        assertTrue(ClinicalEvidenceFunctions.hasHigherOrEqualEvidenceForEventAndTreatmentApproach(createProtectEvidenceList(EvidenceLevel.C,
```
Comment (mdenburger) [inline patient-reporter/src/test/java/com/hartwig/oncoact/patientreporter/cfreport/chapters/analysed/ClinicalEvidenceFunctionsTest.java:41]: A lot of test cases have been deleted. Why?

Comment (LiekeSchoenmaker) [reply patient-reporter/src/test/java/com/hartwig/oncoact/patientreporter/cfreport/chapters/analysed/ClinicalEvidenceFunctionsTest.java:41]: I can revert it, but it tests the same

**Review by mvanniekerkHartwig — APPROVED** [review]

**Review by mdenburger — APPROVED** [review]

**Pull request merged**